src/h5web/visualizations/shared/AxisSystemProvider.tsx

```diff
@@ -1,51 +1,34 @@
 import React, { ReactElement, ReactNode, createContext } from 'react';
-import { scaleLinear } from 'd3-scale';
-import { AxisConfig, AxisInfo, ScaleType, SCALE_FUNCTIONS } from './models';
-import { isIndexAxisConfig } from './utils';
+import {
+  AxisConfig,
+  ScaleType,
+  IndexAxisConfig,
+  DataAxisConfig,
+} from './models';
 
 export interface AxisConfigs {
-  abscissaInfo: AxisInfo;
-  ordinateInfo: AxisInfo;
+  abscissaConfig: AxisConfig;
+  ordinateConfig: AxisConfig;
 }
 
 export const AxisSystemContext = createContext<AxisConfigs>({} as AxisConfigs);
 
-function getAxisInfo(config: AxisConfig): AxisInfo {
-  if (isIndexAxisConfig(config)) {
-    const { indexDomain, showGrid = false } = config;
-    return {
-      isIndexAxis: true,
-      scaleFn: scaleLinear,
-      domain: indexDomain,
-      scaleType: ScaleType.Linear,
-      showGrid,
-    };
-  }
-
-  const { dataDomain, scaleType = ScaleType.Linear, showGrid = false } = config;
-  return {
-    isIndexAxis: false,
-    scaleFn: SCALE_FUNCTIONS[scaleType],
-    domain: dataDomain,
-    scaleType,
-    showGrid,
-  };
-}
-
 type Props = {
-  abscissaConfig: AxisConfig;
-  ordinateConfig: AxisConfig;
+  abscissaConfig: IndexAxisConfig | DataAxisConfig;
+  ordinateConfig: IndexAxisConfig | DataAxisConfig;
   children: ReactNode;
 };
 
 function AxisSystemProvider(props: Props): ReactElement {
   const { abscissaConfig, ordinateConfig, children } = props;
 
-  const abscissaInfo = getAxisInfo(abscissaConfig);
-  const ordinateInfo = getAxisInfo(ordinateConfig);
-
   return (
-    <AxisSystemContext.Provider value={{ abscissaInfo, ordinateInfo }}>
+    <AxisSystemContext.Provider
+      value={{
+        abscissaConfig: { ...abscissaConfig, scaleType: ScaleType.Linear },
+        ordinateConfig: { ...ordinateConfig, scaleType: ScaleType.Linear },
```

Comment:
Hmmm, you're always overriding it like this. You need to destructure the config after the defaults. Perhaps add a `DEFAULT_AXIS_CONFIG` at the top of the file as well, with `scaleType` and perhaps also `isIndexAxis: false` for clarity?